### ISSUE TITLE
Lazy loading doesn't work with Angular 2 spa.

### ISSUE THREAD

**Issue by j-nord** [aspnet/JavaScriptServices#820]
We use Angular 2 template but it is not possible to use lazy loading with this template.

What ist wrong?

                { test: /\.ts$/, include: /ClientApp/, use: ['awesome-typescript-loader', 'angular2-template-loader', 'angular-router-loader'] },

Call:
        path: 'login', loadChildren: './+login/login.module#LoginModule',

**Comment by seven1986** [comment]
see here https://github.com/aspnet/JavaScriptServices/issues/792

**Comment by j-nord** [comment]
Sorry #792 doesn't help. We have all search.  Nothing runs. Debugging breaks with lazy loading modules.  So we can't work. We search one bug after next.

Is there a final Version which run really with lazy-loading and Debugging?

**Comment by SteveSandersonMS** [comment]
@j-nord I don't know what's not working in your case, but I just tried adding lazy loading using the exact set of changes in the example mentioned from #792, and it worked perfectly.

I've just pushed an update to this example so that it's now rebased onto the latest template: https://github.com/aspnet/JavaScriptServices/commit/df5d27d0d8fb82570cb5d72ce7b763601ab3b64b. This definitely does lazy loading. So, I'd recommend you use that as a basis for comparing against your application. Hope that's OK!
